resolver/retry_resolver.go

```diff
@@ -26,11 +26,25 @@ func NewRetryResolver(resolver Resolver, backoffs []time.Duration) Resolver {
 }
 
 func (r *retryResolver) retry(fn func() error) (err error) {
+	var dnsError *net.DNSError
 	err = fn()
-	for i := 0; i < len(r.backoffs) && err != nil && isTemporary(err); i++ {
+
+	for i := 0; i < len(r.backoffs) && errors.As(err, &dnsError) && dnsError.Temporary(); i++ {
 		time.Sleep(r.backoffs[i])
 		err = fn()
 	}
+
+	if errors.As(err, &dnsError) && dnsError.Err == "server misbehaving" {
```

Comment:
Using a string check here, because that's the only thing available I could find. Other ideas welcome.